src/buffer/out/textBuffer.hpp

```diff
@@ -330,16 +295,17 @@ class TextBuffer final
     std::vector<til::point_span> SearchText(const std::wstring_view& needle, bool caseInsensitive) const;
     std::vector<til::point_span> SearchText(const std::wstring_view& needle, bool caseInsensitive, til::CoordType rowBeg, til::CoordType rowEnd) const;
 
-    const std::vector<ScrollMark>& GetMarks() const noexcept;
+    // Mark handling
+    std::vector<ScrollMark> GetMarkRows() const;
+    std::vector<MarkExtents> GetMarkExtents(std::optional<size_t> limit = std::nullopt) const;
     void ClearMarksInRange(const til::point start, const til::point end);
-    void ClearAllMarks() noexcept;
-    void ScrollMarks(const int delta);
-    void StartPromptMark(const ScrollMark& m);
-    void AddMark(const ScrollMark& m);
-    void SetCurrentPromptEnd(const til::point pos) noexcept;
-    void SetCurrentCommandEnd(const til::point pos) noexcept;
-    void SetCurrentOutputEnd(const til::point pos, ::MarkCategory category) noexcept;
-    std::wstring_view CurrentCommand() const;
+    void ClearAllMarks();
+    std::wstring CurrentCommand() const;
+    std::vector<std::wstring> Commands() const;
+    void StartPrompt();
+    bool StartCommand();
+    bool StartOutput();
+    void EndOutput(std::optional<unsigned int> error);

```

Comment:
what's the difference between `nullopt` and `int{0}` here? One indicates that there was no error (OK) and the other indicates that there was a success (wait a sec)

Reply:
PowerShell can actually tell us what the exit code was, but CMD cannot. So in PowerShell, you can get the green/red marks for success/error, but in CMD, I didn't want to assume success for "no result given"

Reply:
_ah!_ that makes sense